### PR TITLE
Introduce kn-vsphere plugin with login and version subcommands

### PR DESCRIPTION
**Pull request by fbiville**
- update e2e tests to use the plugin
- update release scripts for the plugin

Relates to #60

### CODE REVIEW

**Review by mattmoor — COMMENTED** [review]
Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -path './third_party' -prune -o -type f -name '*.go' -print)`

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -65,3 +65,34 @@ jobs:
...
+          go-version: 1.14.x
+
+      - name: Run GoReleaser
+        uses: goreleaser/goreleaser-action@v1
+        with:
+          version: latest
+          args: release --rm-dist
+          workdir: ./plugins/vsphere
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (mattmoor) [inline .github/workflows/release.yaml:98]: Add trailing newline:
```suggestion
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

```

File: plugins/vsphere/pkg/command/login_test.go [hunk 2]
```diff
@@ -0,0 +1,252 @@
+/*
+Copyright 2020 VMware, Inc.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package command_test
+
+import (
+	"fmt"
+	"io/ioutil"
```
Comment (mattmoor) [inline plugins/vsphere/pkg/command/login_test.go:10]: Format Go code:
```suggestion
	"io/ioutil"
	"strings"
	"testing"

```

File: plugins/vsphere/pkg/command/login_test.go [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+	"k8s.io/apimachinery/pkg/runtime"
+	k8stesting "k8s.io/client-go/testing"
+	"k8s.io/client-go/tools/clientcmd"
+	"strings"
+	"testing"
```
Comment (mattmoor) [inline plugins/vsphere/pkg/command/login_test.go:15]: Format Go code:
```suggestion

```

File: plugins/vsphere/pkg/command/version_test.go [hunk 4]
```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2020 VMware, Inc.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package command_test
+
+import (
+	"bytes"
+	"fmt"
```
Comment (mattmoor) [inline plugins/vsphere/pkg/command/version_test.go:10]: Format Go code:
```suggestion
	"fmt"
	"testing"

```

File: plugins/vsphere/pkg/command/version_test.go [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+	"github.com/vmware-tanzu/sources-for-knative/plugins/vsphere/pkg/command"
+	"testing"
```
Comment (mattmoor) [inline plugins/vsphere/pkg/command/version_test.go:12]: Format Go code:
```suggestion

```

**Review by mattmoor — COMMENTED** [review]
Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -path './third_party' -prune -o -type f -name '*.go' -print)`

File: .github/workflows/release.yaml (hunk 1 above)
Comment (mattmoor) [inline .github/workflows/release.yaml:98]: Add trailing newline:
```suggestion
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

```

File: plugins/vsphere/pkg/command/login_test.go (hunk 2 above)
Comment (mattmoor) [inline plugins/vsphere/pkg/command/login_test.go:10]: Format Go code:
```suggestion
	"io/ioutil"
	"strings"
	"testing"

```

**Review by mattmoor — COMMENTED** [review]
Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -path './third_party' -prune -o -type f -name '*.go' -print)`
  `goimports -w $(find -name '*.go' | grep -v vendor | grep -v third_party | grep -v wire_gen.go)`

File: plugins/vsphere/pkg/command/login_test.go (hunk 3 above)
Comment (mattmoor) [inline plugins/vsphere/pkg/command/login_test.go:15]: Format Go code:
```suggestion

```

File: plugins/vsphere/pkg/command/version_test.go (hunk 4 above)
Comment (mattmoor) [inline plugins/vsphere/pkg/command/version_test.go:10]: Format Go code:
```suggestion
	"fmt"
	"testing"

```

File: plugins/vsphere/pkg/command/version_test.go (hunk 5 above)
Comment (mattmoor) [inline plugins/vsphere/pkg/command/version_test.go:12]: Format Go code:
```suggestion

```

File: .github/workflows/release.yaml (hunk 1 above)
Comment (mattmoor) [inline .github/workflows/release.yaml:98]: Add trailing newline:
```suggestion
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

```

File: plugins/vsphere/pkg/command/login_test.go (hunk 2 above)
Comment (mattmoor) [inline plugins/vsphere/pkg/command/login_test.go:10]: Format Go code:
```suggestion
	"io/ioutil"
	"strings"
	"testing"

```

**Review by mattmoor — COMMENTED** [review]
Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -path './third_party' -prune -o -type f -name '*.go' -print)`
  `goimports -w $(find -name '*.go' | grep -v vendor | grep -v third_party | grep -v wire_gen.go)`

File: plugins/vsphere/pkg/command/login_test.go (hunk 3 above)
Comment (mattmoor) [inline plugins/vsphere/pkg/command/login_test.go:15]: Format Go code:
```suggestion

```

File: plugins/vsphere/pkg/command/version_test.go (hunk 4 above)
Comment (mattmoor) [inline plugins/vsphere/pkg/command/version_test.go:10]: Format Go code:
```suggestion
	"fmt"
	"testing"

```

File: plugins/vsphere/pkg/command/version_test.go (hunk 5 above)
Comment (mattmoor) [inline plugins/vsphere/pkg/command/version_test.go:12]: Format Go code:
```suggestion

```

File: .github/workflows/release.yaml [hunk 6]
```diff
@@ -65,3 +65,37 @@ jobs:
...
+          go-version: 1.14.x
+
+      - name: Run GoReleaser
+        uses: goreleaser/goreleaser-action@v1
+        with:
+          version: latest
+          args: release --rm-dist
+          workdir: ./src/github.com/${{ github.repository }}/plugins/vsphere
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (mattmoor) [inline .github/workflows/release.yaml:100]: One lurking concern I have is that we'll end up with two releases from these two jobs.  Something we'll want to check on post-submit.

File: plugins/vsphere/pkg/command/login.go [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+		Short: "Create the required vSphere credentials",
+		Long: `Create the required vSphere credentials
+
+Examples:
+# Log in the default namespace
+kn vsphere login --username jane-doe --password s3cr3t --secret-name vsphere-credentials
+# Log in the specified namespace
+kn vsphere login --namespace ns --username john-doe --password s3cr3t --secret-name vsphere-credentials
+# Log in the specified namespace with the password retrieved via standard input
+kn vsphere login --namespace ns --username john-doe --password-stdin --secret-name vsphere-credentials
```
Comment (mattmoor) [inline plugins/vsphere/pkg/command/login.go:43]: There is an "Example" field for these, right?

**Review by mattmoor — DISMISSED** [review]

File: plugins/vsphere/pkg/command/login.go (hunk 7 above)
Comment (fbiville) [reply plugins/vsphere/pkg/command/login.go:43]: Oops, yes! 😅

File: .github/workflows/release.yaml (hunk 6 above)
Comment (fbiville) [reply .github/workflows/release.yaml]: is there a way to tell `ko` to ignore `./plugins/*`? That way, plugins would be released separately.

**Comment by fbiville** [comment]
Currently struggling with goreleaser. I opened https://github.com/goreleaser/goreleaser-action/issues/223 for guidance.

File: .goreleaser.yml [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+  sort: asc
+  filters:
+    exclude:
+    - '^docs:'
+    - '^test:'
+# drop the lines below once PR is merged
+release:
+  github:
+    owner: fbiville
+    name: sources-for-knative
```
Comment (mattmoor) [inline .goreleaser.yml:40]: Add trailing newline:
```suggestion
    name: sources-for-knative

```

File: .goreleaser.yml [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
```
Comment (mattmoor) [inline .goreleaser.yml:43]: Add trailing newline:
```suggestion
    name: sources-for-knative

```

File: .goreleaser.yml [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
```
Comment (mattmoor) [inline .goreleaser.yml:45]: Add trailing newline:
```suggestion
    name: sources-for-knative

```

File: .goreleaser.yml [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
```
Comment (mattmoor) [inline .goreleaser.yml:44]: Add trailing newline:
```suggestion
    name: sources-for-knative

```

File: .github/workflows/release.yaml (hunk 6 above)
Comment (fbiville) [reply .github/workflows/release.yaml]: I think the current release script for the whole repo can be ported to goreleaser, as a build step preceding the plugin build step. I'll have a crack at it tomorrow.

Comment (fbiville) [reply .github/workflows/release.yaml]: So I tried using gorelease for the ko release but it's a dead-end.
What I'm doing instead is have the ko release job depend on the plugin release job.
The ko release job then piggybacks on the release created by goreleaser to upload the `release.yaml`.

**Review by mattmoor — APPROVED** [review]

**Pull request merged**